frontend/src/components/data-table/columns.tsx

```diff
@@ -115,12 +235,15 @@ export function generateColumns<T>({
         }
 
         const value = getValue();
+        const dtype = fieldTypes?.[info.key]?.[1];
+
         if (isPrimitiveOrNullish(value)) {
           const rendered = renderValue();
           if (rendered == null) {
             return "";
           }
-          return String(rendered);
+          // return String(rendered);
+          return applyFormat(rendered, columnFormats[info.key], dtype);
```

Comment:
i think this formatting should exist on the backend, instead of the frontend. this is a nice touch since it can do per-column, but maybe we can stash this code for later. doing this on the backend will make the `Download As` also formatted. 

I think the formatting could be right before we call `to_csv` in the `*TableManager` (e.g. `PolarsTableManager`)

```python
@dataclass
class FormattingOptions
      datetime_format: str | None = None
      date_format: str | None = None
      time_format: str | None = None
      float_scientific: bool | None = None
      float_precision: int | None = None
      null_value: str | None = None # maybe?
```
   
for polars they expose the options on write_csv (and json) that we can pass through
see https://docs.pola.rs/api/python/stable/reference/api/polars.DataFrame.write_csv.html

for pandas we can set the context before writing 
https://pandas.pydata.org/docs/reference/api/pandas.option_context.html

Reply:
actually - we may not want to format the `download as` so it can be re-imported, without unformatting.

Reply:
Yea makes sense for `download as` to download the raw data